torchdynamo/variables/torch.py

```diff
@@ -158,6 +158,10 @@ def call_function(
             and args[0].size is not None
         ):
             return ConstantVariable(product(args[0].size), **options)
+        elif self.value is torch._shape_as_tensor:
+            assert len(args) == 1
+            log.warning("torch._shape_as_tensor(x) is going to be rewritten as x.size()")
+            return args[0].call_method(tx, "size", [], {})
```

Comment:
cc @Chillee - was this the dynamo re-write you had in mind?

(Looks roughly right to me but I'm not thaat familiar with dynamo internals :) )